### ISSUE TITLE
Multimedia Player - Video: Support for Internet Explorer 8 dropped?

### ISSUE THREAD

**Issue by Christopher-O** [wet-boew/wet-boew#6645]
I noticed that between WET 4.0.11 and WET 4.0.12 that the media videos on the GitHub page, and a couple of testing pages I was working are no longer viewable in Internet Explorer 8.  On the GitHub page (MultiMedia player - Video), the images do not show up.  On my local testing page, the image shows up, but no media controls show up at the bottom.  If I switch the files to WET 4.0.11 the media player works just fine in Internet Explorer 8.  Is there any fix to this issue or has support dropped?

**Comment by Christopher-O** [comment]
Sorry, just a further update.  On the Github page, when I click on the "Compatibility View" button, the images then show up, but the videos still do not have any media control at the bottom to run the video.

**Comment by cfarquharson** [comment]
I just put the intranet theme on our intranet and although the WET 4.0.12 muti-media player works in Firefox, it doesn't work in IE11.  It works in IE11 on github but not in the download.  Is there something missing form the build?

**Comment by pjackson28** [comment]
That's probably because your IE11 is configured to run in compatibility mode when you are in the Intranet zone versus Edge mode when in the Internet zone (we don't support compatability mode). You can find some help for ensuring Edge mode gets triggered here: http://wet-boew.github.io/wet-boew/docs/server-en.html

**Comment by cfarquharson** [comment]
@pjackson28 no, it's in Edge mode

**Comment by cfarquharson** [comment]
@pjackson28 further to the above, every other script solution works in IE11, it's just the multimedia player that doesn't.

**Comment by cfarquharson** [comment]
@pjackson28 also, it works in IE11 on WET4.0.11, just not 12

**Comment by pjackson28** [comment]
@cfarquharson I tried the WET theme and Intranet theme multimedia player examples in IE11 and they worked fine for me. I checked and both have v4.0.12 in the comment block. Do the multimedia players at either of these links work for you?
* http://wet-boew.github.io/v4.0-ci/demos/multimedia/multimedia-en.html
* http://wet-boew.github.io/themes-dist/theme-gc-intranet/demos/multimedia/multimedia-en.html

Also how are you running your test with the downloaded files? Are you opening them through File Explorer or are you running them through an actual web server? Also, are you getting an errors in your error console?

**Comment by LaurentGoderre** [comment]
Yeah, it seem that the player is broken in IE8

**Comment by cfarquharson** [comment]
@pjackson28 We put the files on our intranet server.  The videos work in IE11 when I view the files on github but the exact same files on our intranet don't work in 4.0.12 (again, all good in 4.0.11).  As @LaurentGoderre confirms, there is now no support for IE8.  IE11 is in Edge mode so I'm not sure if there is a problem with the download package missing something or if it's tied to the same reason that IE8 no longer lots works at all with the multimedia player.

**Comment by LaurentGoderre** [comment]
Can you paste the code after the page has finished loading for the video in 4.0.11?

**Comment by LaurentGoderre** [comment]
Can you paste the one for v4.0.12

**Comment by cfarquharson** [comment]
Below are the IE11 and IE12 generated code from Developer Tools (F12)

Here's the working IE11 generated code: 
```
<figure class="wb-mltmd wb-mltmd-inited video" id="wb-auto-9">
<!-- Web Experience Toolkit (WET) / Boîte à outils de l'expérience Web (BOEW)
wet-boew.github.io/wet-boew/License-en.html / wet-boew.github.io/wet-boew/Licence-fr.html -->
<!-- DataAjaxFragmentStart -->
<div class="display">
<button class="wb-mm-ovrly" aria-controls="wb-auto-9-md" type="button" data-role="none"><span class="wb-inv">Play</span></button><object width="555" height="312" tabindex="-1" class="video" id="wb-auto-9-md" data="http://workzone/WET4/development/cxo149/wet411/js/../assets/multimedia.swf?1427378434869" type="application/x-shockwave-flash" style="height: 312px;" pause="" play=""><param name="movie" value="http://workzone/WET4/development/cxo149/wet411/js/../assets/multimedia.swf?1427378434869"><param name="flashvars" value="id=wb-auto-9-md&amp;media=http://video2.servicecanada.gc.ca/video/boew-wet/te-lj-eng.mp4&amp;height=312&amp;width=555&amp;posterimg=http://www.servicecanada.gc.ca/eng/video/images/te-lj-eng.jpg"><param name="allowScriptAccess" value="always"><param name="bgcolor" value="#000000"><param name="wmode" value="opaque"><img width="555" height="312" class="img-responsive" alt="Looking for a Job" src="http://www.servicecanada.gc.ca/eng/video/images/te-lj-eng.jpg"></object>
</div>
<div class="wb-mm-cc">&nbsp;</div>
<div class="wb-mm-ctrls">
<div class="frstpnl">
<div class="btn-group">
<button title="Play" class="btn btn-default playpause" aria-controls="wb-auto-9-md" type="button" data-state-off="Pause" data-state-on="Play"><span class="glyphicon glyphicon-play"><span class="wb-inv">Play</span></span></button>
<button title="Mute" class="btn btn-default mute" aria-pressed="false" aria-controls="wb-auto-9-md" type="button" data-state-off="Unmute" data-state-on="Mute"><span class="glyphicon glyphicon-volume-up"><span class="wb-inv">Mute</span></span></button>
<input title="Volume" class="volume" aria-controls="wb-auto-9-md" type="range" min="0" max="100" step="5" value="100">
</div>
</div>
<div class="tline">
<div class="wb-mm-txtonly">
<p class="wb-mm-tmln-crrnt"><span class="wb-inv">Current position:</span><span>00:00:00</span></p>
<p class="wb-mm-tmln-ttl"><span class="wb-inv">Total time:</span><span>--:--:--</span></p>
</div>
<div class="wb-mm-prgrss">
<progress tabindex="0" aria-live="off" value="0" max="100"></progress>
</div>
</div>
<div class="lastpnl">
<div class="btn-group">
<button title="Show closed captioning" class="btn btn-default cc" aria-pressed="false" aria-controls="wb-auto-9-md" type="button" data-state-off="Hide closed captioning" data-state-on="Show closed captioning"><span class="glyphicon glyphicon-subtitles"><span class="wb-inv">Show closed captioning</span></span></button>
</div>
</div>
</div>
<!-- DataAjaxFragmentEnd -->

	<figcaption>
		<details id="inline-captions">
			<summary tabindex="0" class=" wb-details-inited" role="button" aria-expanded="false">Looking for a Job - HTML5 Transcript/Captions</summary>
			<p class="wet-boew-vd"><strong>(Animated pen draws a red line that leads into the text Looking for A Job)</strong></p>
			<p class="wet-boew-vd"><strong>(Cut to a medium shot of the Host. LINE DRAWING GRAPHIC: Stick person appears and waves.)</strong></p>
			<p><span class="wb-tmtxt" data-dur="3.84s" data-begin="2.50s">Hi, my name is Eric, and I'm a Service Canada employee.</span></p>
			...
		</details>
	</figcaption>
</figure>
```
Here's the NON-working IE12 generated code: 

```
<figure class="wb-mltmd wb-mltmd-inited video" id="wb-auto-9">
	<object width="554" height="312" tabindex="-1" class="video" id="wb-auto-9-md" data="http://workzone/WET4/development/cxo149/wet412/js/../assets/multimedia.swf?1427378496516" type="application/x-shockwave-flash" pause="" play=""><param name="movie" value="http://workzone/WET4/development/cxo149/wet412/js/../assets/multimedia.swf?1427378496516"><param name="flashvars" value="id=wb-auto-9-md&amp;media=http://video2.servicecanada.gc.ca/video/boew-wet/te-lj-eng.mp4&amp;height=312&amp;width=554&amp;posterimg=http://www.servicecanada.gc.ca/eng/video/images/te-lj-eng.jpg"><param name="allowScriptAccess" value="always"><param name="bgcolor" value="#000000"><param name="wmode" value="opaque"><img width="554" height="312" class="img-responsive" alt="Looking for a Job" src="http://www.servicecanada.gc.ca/eng/video/images/te-lj-eng.jpg"></object>
	<figcaption>
		<details id="inline-captions">
			<summary tabindex="0" class=" wb-details-inited" role="button" aria-expanded="false">Looking for a Job - HTML5 Transcript/Captions</summary>
			<p class="wet-boew-vd"><strong>(Animated pen draws a red line that leads into the text Looking for A Job)</strong></p>
			<p class="wet-boew-vd"><strong>(Cut to a medium shot of the Host. LINE DRAWING GRAPHIC: Stick person appears and waves.)</strong></p>
			<p><span class="wb-tmtxt" data-dur="3.84s" data-begin="2.50s">Hi, my name is Eric, and I'm a Service Canada employee.</span></p>
			...
		</details>
	</figcaption>
</figure>
```

**Comment by LaurentGoderre** [comment]
I don't know what kind of twisted network configuration that would disable the native support for HTML5 video in IE11 but it seems like your network is. While I need to fix the Flash fallback, you should look into what causes this on your end because this is deteriorating your user's experience.

**Comment by cfarquharson** [comment]
@LaurentGoderre is that what 4.0.12 does, rely on native support whereas 4.0.11 used js to play the video?

**Comment by crichard** [comment]
4.0.12 works fine for me a regular Internet connection in IE11.  
On the DWAN network (IE9), 4.0.12 videos don't load (except in lightboxes, where I trigger the player manually).  4.0.11 works fine on the DWAN.

**Comment by LaurentGoderre** [comment]
@cfarquharson  IE11 should never had used the Flash fallback. Something on your network is wrong for the native video playback to be disabled. WET since v4.0.0 has relied on native and polyfilled browser that didn't have native support.

This issue should be limited to IE8 and Firefox on Windows XP.

**Comment by LaurentGoderre** [comment]
@crichard your IE9 load the player in Flash?

**Comment by cfarquharson** [comment]
@LaurentGoderre we wanted to upgrade to the newest code but can't if it won't play in IE8, as we still have to support a handful of users in IE8 even though most of us are in IE11.  Is IE8 something you can fix asap? (long story on the urgency).

Apparently the video is working on IE11 for some others on our intranet so I have an email into IT to find out why it's not working for me.

**Comment by LaurentGoderre** [comment]
I am trying to get resolved this today. It's probably a very minor thing.

**Comment by crichard** [comment]
4.0.11 in DWAN IE9: Flash is loaded by default, but my manually loaded lightbox videos load HTML5

**Comment by LaurentGoderre** [comment]
We still haven't fix that issue???

**Comment by crichard** [comment]
I thought we had, but it looks like it might have come back?  We fixed the issue for the lightbox, maybe not the in-page videos (#6188).  4.0.10 also loads the Flash in IE9 on DWAN.

**Comment by LaurentGoderre** [comment]
@crichard, I wonder if the issue is that the DWAN slows the processing of resource so much that it thinks it doesn't support any of the source when it fact it simply didn't reply yet.

**Comment by crichard** [comment]
That's a good possibility.  They like to scan everything and strip as many scripts as possible.  I guess that's why the lightbox works, since we added a timeout.  I doubt there is anything we can do about that.

**Comment by LaurentGoderre** [comment]
There might be but I'm afraid to break it for everyone else.

**Comment by LaurentGoderre** [comment]
@Christopher-O @cfarquharson would you be able to use the development release after I merge the fix. The fix would be the only change since v4.0.12

**Comment by cfarquharson** [comment]
@LaurentGoderre how do I get this code? Do you just download the latest Developmental Release of the Intranet theme? http://wet-boew.github.io/wet-boew/docs/versions/dwnld-en.html#v40dev

**Comment by LaurentGoderre** [comment]
The intranet has not been updated yet but it will be in less than an hour

**Comment by cfarquharson** [comment]
@LaurentGoderre but will I download the official release or the Developmental Release?

**Comment by LaurentGoderre** [comment]
Development release

**Comment by Christopher-O** [comment]
Just an update (since I'm sure you are still working on this), I tried the developmental release, and now the video controls work in IE8, but the play button image has been replaced by the number 1 and a slightly smaller 2 (it shows up fine on my IE11).  

Also I do not know if it's something currently being worked on but the page I am testing does not seem to load in Firefox.  The content displays for a second, and then disappears and  is replaced with just the white background and the page seems to continuously load.

**Comment by LaurentGoderre** [comment]
@Christopher-O Can you add a screenshot of this? I suspect we will need to get rid of the playback icon for IE8.

With regards to the other problem, can you reproduce in our working examples? If not can you add a link or the source code?

**Comment by Christopher-O** [comment]
![videoimage](https://cloud.githubusercontent.com/assets/9367162/6855441/d6a87f48-d3bd-11e4-873f-06820ce5266c.jpg)

there is the video screen shot.  As for the firefox issue, I will try to reproduce the examples and see if its something I missed or overwrote.

**Comment by LaurentGoderre** [comment]
Is it possible your server is not serving the file in UTF8?? The two characters makes me think it's incorrectly interpreting the CSS in ANSI instead of UTF8

**Comment by Christopher-O** [comment]
I'm not sure. Its all works properly in IE11 and when we tested 4.0.11, so I unless something new has changed I do not see why it would interpret the symbol differently.  Also I tried using the Multimedia page downloaded in the template.  The file works in IE11, but the video does not show up in in IE8, there is a message that says this "this page could not be displayed".  And when I try to view it in Firefox, I get the exact same blank screen with a simple hyperlink that goes back to the homepage of the server, and the browser does not stop running.

**Comment by LaurentGoderre** [comment]
The two issues are probably unrelated. I will investigate when I get the code or link to a page with the problem

**Comment by tinazl** [comment]
When I start the video, it always display the "Play" sign (trangle) in the middle of video. and it can't display poster for IE. I use version 4.0.17.

**Comment by LaurentGoderre** [comment]
This was fixed in v4.0.18

**Comment by tinazl** [comment]
The download link of v4.0.18 doesn't work. It display "404 - This is not the web page you are looking for" error message.
